x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/FallbackClaim.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+
+import java.text.ParseException;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.core.Strings.format;
+
+public class FallbackClaim {
+    private final String name;
+    private final JWTClaimsSet claimsSet;
+    private final String actualName;
+
+    public FallbackClaim(String name, Map<String, String> fallbackClaimNames, JWTClaimsSet claimsSet) {
```

Comment:
nit: maybe name ClaimWithFallback ? (since the fallback can be null)

Reply:
I agree the name is not great. I renamed it to `FallbackableClaim` which was in fact the original name that I had. But IntelliJ warned me that `Fallbackable` was not a proper word. I think the intention is clearer with it. I'll also add a javadoc for this class. Let me know if this works for you.